lib/experimental/collab.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Functions to support collaboration.
+ *
+ * @package gutenberg
+ */
+
+if ( ! function_exists( 'register_post_meta_for_collab_comment' ) ) {
+	/**
+	 * Registers the collab meta field required for comments to work.
+	 *
+	 * @since // TODO: Add version number.
+	 */
+	function register_post_meta_for_collab_comment() {
+		$post_types = get_post_types( array( 'show_in_rest' => true ) );
+
+		foreach ( $post_types as $post_type ) {
```

Comment:
We might be doing a bunch of unnecessary iterations here. 

You could do it with a more advanced `get_post_types` query (untested):

```suggestion
		$post_types = get_post_types(
			array(
				'show_in_rest' => true,
				 'supports' => array( 'editor', 'custom-fields', 'revisions' ),
			)
		);

		foreach ( $post_types as $post_type ) {
```

Reply:
`get_post_types` function does not have "supports" parameter to it's argument as refer to its [documentation](https://developer.wordpress.org/reference/functions/get_post_types/).

Reply:
The documentation definitely needs a bit of love, but with `get_post_types()` you can filter by any of the `register_post_type()` fields. That's what `wp_filter_object_list` does under the hood - it can filter by any of the object fields. Let me know if that helps.